examples/MagicWeather/app/src/main/java/com/revenuecat/sample/ui/paywall/PaywallAdapter.kt

```diff
@@ -35,19 +38,90 @@ class PaywallAdapter(
         val view = LayoutInflater.from(viewGroup.context)
             .inflate(R.layout.paywall_item, viewGroup, false)
 
-        return PackageViewHolder(view, didChoosePackage)
+        return PackageViewHolder(view, didChoosePaywallItem)
     }
 
     override fun onBindViewHolder(viewHolder: PackageViewHolder, position: Int) {
-        offering?.availablePackages?.get(position)?.let {
+        theItems[position].let {
             val item = it
             viewHolder.item = item
 
-            viewHolder.view.findViewById<TextView>(R.id.paywall_item_title).text = item.product.title
-            viewHolder.view.findViewById<TextView>(R.id.paywall_item_description).text = item.product.description
-            viewHolder.view.findViewById<TextView>(R.id.paywall_item_price).text = item.product.originalPrice
+            when (it) {
+                is PaywallItem.Title -> {
+                    viewHolder.view.findViewById<TextView>(R.id.paywall_item_options_title).visibility = View.VISIBLE
+                    viewHolder.view.findViewById<TextView>(R.id.paywall_item_options_title).text = it.title
+                    viewHolder.view.findViewById<ConstraintLayout>(R.id.purchasableLayout).visibility = View.GONE
+                }
+                is PaywallItem.Product -> {
+                    viewHolder.view.findViewById<TextView>(R.id.paywall_item_options_title).visibility = View.GONE
+                    viewHolder.view.findViewById<ConstraintLayout>(R.id.purchasableLayout).visibility = View.VISIBLE
+
+                    val product = it.storeProduct
+                    val price = product.oneTimeProductPrice?.formattedPrice ?: "???"
+
+                    viewHolder.view.findViewById<TextView>(R.id.paywall_item_price).text = price
+                    viewHolder.view.findViewById<TextView>(R.id.paywall_item_best_offer).visibility = View.GONE
+                }
+                is PaywallItem.Option -> {
+                    viewHolder.view.findViewById<TextView>(R.id.paywall_item_options_title).visibility = View.GONE
+                    viewHolder.view.findViewById<ConstraintLayout>(R.id.purchasableLayout).visibility = View.VISIBLE
+
+                    val option = it.subscriptionOption
+                    val price = option.pricingPhases.map { phase ->
+                        "${phase.formattedPrice} for ${phase.billingPeriod}"
+                    }.joinToString(separator = " -> ")
+
+                    viewHolder.view.findViewById<TextView>(R.id.paywall_item_price).text = price
```

Comment:
maybe we can pull these lines into a method like `setUpOptionView` or something?